src/base-test-class.js

```diff
@@ -122,7 +122,20 @@ BaseTest.prototype = {
               failed: o.failed
             };
           })
-        : null;
+        : {};
+
+        for(let key in this){
+          const val = this[key];
+          if(typeof val === 'function' && !['before', 'after', 'beforeEach', 'afterEach'].includes(key)){
```

Comment:
you don't want `includes` here - you want straight equality - otherwise you skip functions named: 'check this happens before that' ?

Reply:
Are you sure? According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes, it's like Java's indexOf() function where it has to be a complete match

Reply:
Without nightwatch-extra's base class, you lose any correlation to SauceLabs videos and anything that requires the sessionId. There is a way to do this without the base class, but it will require a lot of changes and a lot of work. I'm just using what we already have in place.

Reply:
ah you're right i was thinking of String.includes!